bus-mapping/src/external_tracer.rs

```diff
@@ -1,78 +1,90 @@
+    origin: Address,
+    gas_limit: Word,
+    target: Address,
 }
 
-impl<F: FieldExt> Default for Transaction<F> {
-    fn default() -> Self {
+impl Transaction {
+    /// Create Self from a web3 transaction
+    pub fn from_eth_tx(tx: &eth_types::Transaction) -> Self {
+        Self {
+            origin: tx.from.unwrap(),
+            gas_limit: tx.gas,
+            target: tx.to.unwrap(),
+        }
+    }
+
+    /// Generate a new mock Self with preloaded data, useful for tests.
+    pub fn mock() -> Self {
         Transaction {
-            origin: EthAddress::from_str(
-                "0x00000000000000000000000000000000c014ba5e",
-            )
-            .unwrap(),
-            gas_limit: F::from_u64(1_000_000u64),
-            target: EthAddress::zero(),
+            origin: crate::address!(
+                "0x00000000000000000000000000000000c014ba5e"
+            ),
+            gas_limit: Word::from(1_000_000u64),
+            target: Address::zero(),
         }
     }
 }
 
 /// Definition of all of the data related to an account.
 #[derive(Debug, Clone, Serialize)]
-pub struct Account<F: FieldExt> {
-    address: EthAddress,
-    #[serde(serialize_with = "serialize_field_ext")]
-    balance: F,
+pub struct Account {
+    address: Address,
+    balance: Word,
     code: String,
 }
 
+impl Account {
+    /// Generate a new mock Self with preloaded data, useful for tests.
+    pub fn mock(code: &Bytecode) -> Self {
+        Self {
+            address: Transaction::mock().target,
+            balance: Word::from(555u64),
+            code: hex::encode(code.to_bytes()),
+        }
+    }
+}
+
 #[derive(Debug, Clone, Serialize)]
-#[serde(bound(serialize = "BlockConstants<F>: Serialize",))]
-struct GethConfig<F: FieldExt> {
-    block_constants: BlockConstants<F>,
-    transaction: Transaction<F>,
-    accounts: Vec<Account<F>>,
+struct GethConfig {
+    block_constants: BlockConstants,
+    transaction: Transaction,
+    accounts: Vec<Account>,
 }
 
 /// Creates a trace for the specified config
-pub fn trace<F: FieldExt>(
-    block_constants: &BlockConstants<F>,
-    code: &Bytecode,
-) -> Result<Vec<ExecutionStep>, Error> {
-    // Some default values for now
-    let transaction = Transaction::default();
-    let account = Account {
-        address: transaction.target,
-        balance: F::from_u64(555u64),
-        code: hex::encode(code.to_bytes()),
-    };
-
+pub fn trace(
+    block_constants: &BlockConstants,
+    tx: &Transaction,
+    accounts: &[Account],
+) -> Result<Vec<GethExecStep>, Error> {
     let geth_config = GethConfig {
         block_constants: block_constants.clone(),
-        transaction,
-        accounts: vec![account],
+        transaction: tx.clone(),
+        accounts: accounts.to_vec(),
     };
 
     // Get the trace
-    let trace =
+    let trace_string =
         geth_utils::trace(&serde_json::to_string(&geth_config).unwrap())
+            // TODO: Capture error into TracingError if possible
```

Comment:
I'd say we either address this in this PR or we file an issue for it and so we remove the `TODO`.

Reply:
done https://github.com/appliedzkp/zkevm-circuits/issues/165